lib/std/Build.zig

```diff
@@ -94,6 +94,9 @@ available_deps: AvailableDeps,
 
 release_mode: ReleaseMode,
 
+// True only for the top-level builder.
+is_root: bool = false,
+
```

Comment:
As far as I'm aware whichever way this functionality is implemented requires tracking this information, which could be abused by users, for example by making the root builder forget that it's the root one.

Reply:
I agree that while this could be abused, there's no way around it.